envoy/network/proxy_protocol.h

```diff
@@ -1,13 +1,24 @@
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include "envoy/network/address.h"
 
 namespace Envoy {
 namespace Network {
 
+struct ProxyProtocolTLV {
+  const uint8_t type;
+  const std::string value;
+};
+
+using ProxyProtocolTLVVector = std::vector<ProxyProtocolTLV>;
+
 struct ProxyProtocolData {
   const Network::Address::InstanceConstSharedPtr src_addr_;
   const Network::Address::InstanceConstSharedPtr dst_addr_;
+  const ProxyProtocolTLVVector tlv_vector_{};
   std::string asStringForHash() const {
```

Comment:
Do you know what this hash is used for and whether TLVs need to be included?

Reply:
looks like it is used [here](https://github.com/envoyproxy/envoy/blob/main/source/common/network/transport_socket_options_impl.cc#L46) by virtue of [TransportSocketOptions::hashKey](https://github.com/envoyproxy/envoy/blob/main/envoy/network/transport_socket.h#L217): `used to separate connections based on the option`. given that this already functions to "separate connections" in the mainline and TLVs wouldn't add anything to further disambiguate connections that i can see, i think they can be omitted here :-)

Reply:
Thank you!

Reply:
That's a great question. Thanks Greg for investigation!